manifests/gpgkey.pp

```diff
@@ -58,25 +58,26 @@
     mode   => $mode,
   }
 
-  $rpmname = "gpg-pubkey-$(gpg ${path} | head -1 | cut -c12-20 | \
-tr '[A-Z]' '[a-z]')"
+  $keys = yum::get_gpg_keys($path)
 
-  case $ensure {
-    'present', default: {
-      exec { "rpm-import-${name}":
-        path    => '/bin:/usr/bin:/sbin/:/usr/sbin',
-        command => "rpm --import ${path}",
-        unless  => "rpm -q ${rpmname}",
-        require => File[$path],
+  $keys.each |String kkey| {
```

Comment:
just for a test, could you remove the keyword `String` here? I'm not sure if the datatype here is a Puppet 5 only feature.

Reply:
Still getting 
manifests/gpgkey.pp:64:variable_scope:WARNING:top-scope variable being used without an explicit namespace
on first 3 tests

Reply:
I am out of ideas and tired of tinkering.
@bastelfreak  - Any suggestions ?